provider/diag/diagnostics.go

```diff
@@ -126,19 +126,25 @@ func (diags Diagnostics) Squash() Diagnostics {
 }
 
 func (diags Diagnostics) Warnings() uint64 {
-	return diags.CountBySeverity(WARNING)
+	return diags.CountBySeverity(WARNING, false)
 }
 
 func (diags Diagnostics) Errors() uint64 {
-	return diags.CountBySeverity(ERROR)
+	return diags.CountBySeverity(ERROR, false)
 }
 
-func (diags Diagnostics) CountBySeverity(sev Severity) uint64 {
+// CountBySeverity returns number of diagnostics of the given severity. If deep is false, squashed diags are counted as a single diag.
+func (diags Diagnostics) CountBySeverity(sev Severity, deep bool) uint64 {
```

Comment:
don't like the argument deep its ambiguous, maybe `include_squashed` or  `count_squashed` so its obvious by the param name what this flag turns on.

Reply:
fixed